Framework/Algorithms/src/WienerSmooth.cpp

```diff
@@ -366,12 +379,19 @@ WienerSmooth::smoothSingleSpectrum(API::MatrixWorkspace_sptr inputWS,
   // copy the x-values and errors from the original spectrum
   // remove the last values for odd-sized inputs
   if (isOddSize) {
-    out->dataX(0).assign(X.begin(), X.end() - 1);
-    out->dataE(0).assign(E.begin(), E.end() - 1);
-    out->dataY(0).resize(Y.size() - 1);
+    auto histogram = out->histogram(0);
+    auto newSize = histogram.y().size() - 1;
+    histogram.resize(newSize);
+    out->setHistogram(0, histogram);
+
+    auto &xOut = out->mutableX(0);
+    xOut.assign(X.begin(), X.end() - 1);
+    auto &eOut = out->mutableE(0);
+    eOut.assign(E.begin(), E.end() - 1);
```

Comment:
Maybe something like
```cpp
out->setHistogram(0, BinEdges(X.begin(), X.end()-1), Counts(...), CountStandardDeviations(...));
```
Should work, if you know what your histogram is (histogram / points/ distribution)?